internal/interfaces/grpc/service.go

```diff
@@ -489,3 +502,29 @@ func (s *service) startListeningToReadyChan() {
 	s.muxOperator = services.muxOperator
 	s.muxTrade = services.muxTrade
 }
+
+func (s *service) tradeGrpcGateway(ctx context.Context, tlsKey, tlsCert string) (http.Handler, error) {
+	creds := grpc.WithTransportCredentials(insecure.NewCredentials())
+	if tlsKey != "" {
+		tlsConfig, err := getTlsConfig(tlsKey, tlsCert)
+		if err != nil {
+			return nil, err
+		}
+
+		creds = grpc.WithTransportCredentials(credentials.NewTLS(tlsConfig))
+	}
+
+	conn, err := grpc.DialContext(ctx, s.opts.TradeAddress, []grpc.DialOption{creds}...)
```

Comment:
DialContext should be used in combo with a context-with-deadline. Why not using just Dial() here without any ctx?

Reply:
Done